google/ggrc-core
# GGRC-4803 Update evidence url and assessment status in single PUT

# Issue description

Currently there are two problems with update evidence url and status in single PUT:
1) status is reset back to 'In Progress' due to `autostatuschangable` mixin;
2) there is an error in notification handlers (for the scenario described in steps to test changes), which appears because there are 2 hooks which try to prepare notifications. The first hook puts notification into session. The second hook have to perform delete query, and the notification putted into session in the previous hook is have to be deleted too. But it is not deleted from the session, because `synchronize_session=False` is set on delete. This cause an error on flush.

# Steps to test the changes

Currently changes can not be tested on dev as `Complete` button is disabled. The scenario after the button will be unblocked:
1) Open assessment info pane;
2) Add/remove evidence url;
3) Quickly press 'Complete' button;
Actual result: 500 error appears.
Expected result: No error appears, assessment state updated successfully.

# Solution description

Checks if status was changed in put request by user have been added. If status was changed, it doesn't have to be reset. Also, required notifications are deleted from the session manually. 
This helps to avoid an error on flush.

# Sanity checklist




# PR Review checklist
